### ISSUE TITLE
[sklearn#33559] Add multi-threading to DecisionTreeClassifier and DecisionTreeRegressor

### ISSUE THREAD

**Issue by godofecht** [flooooooooooow/flow#390]
**Mirrored from [scikit-learn#33559](https://github.com/scikit-learn/scikit-learn/issues/33559)**

Original title: Add multi-threading to DecisionTreeClassifier and DecisionTreeRegressor
Original labels: New Feature, Needs Decision - Include Feature
Created: 2026-03-16T14:27:09Z

---

### Describe the workflow you want to enable

```python
from sklearn.tree import DecisionTreeClassifier()

tree = DecisionTreeClassifier(n_threads=4)
tree.fit(X, y)
```

### Describe your proposed solution

Use Cython `parange`, in particular in  the sorting of features in the tree partitioner.

### Additional context

Multi-threading has been proposed in #5212 among other things. This issue is to only track multi-threading for trees.

`GradienBoosting*` estimators could also benefit (random forests not so much as they can already profit from learning trees in parallel).

---

## Flow context

This issue is mirrored from scikit-learn to explore whether Flow encounters the same problem or can approach it differently. Add analysis below.

**Comment by godofecht** [comment]
Moved to [6](https://github.com/godofecht/flow-scikit/issues/6) in the flow-scikit repository. Further library-specific triage and implementation will happen there.
